src/main/java/gui/EditRulePopup.java

```diff
@@ -0,0 +1,274 @@
+		metricsPanel.add(metricsScrollpane, BorderLayout.CENTER);
+		return metricsPanel;
+	}
+
+	/**
+	 * @return Returns the JPanel responsible for holding the JButtons which allow
+	 *         all metrics to be cleared, or the rule to be saved.
+	 */
+	private JPanel createControlPanel() {
+		controlPanel = new JPanel();
+		controlPanel.setLayout(new GridLayout(1, 3));
+
+		JButton clearButton = new JButton("Clear Metrics");
+		JButton deleteButton = new JButton("Delete Rule");
+		JButton saveButton = new JButton("Save Rule");
+
+		clearButton.addActionListener(new ActionListener() {
+			public void actionPerformed(ActionEvent e) {
+				clearMetricsListPanel();
+			}
+		});
+
+		deleteButton.addActionListener(new ActionListener() {
+			public void actionPerformed(ActionEvent e) {
+				if (nameText.getText().isEmpty()) {
+					JOptionPane.showMessageDialog(null, "Please insert a rule name!");
+				} else {
+					// Add functionality here.
+				}
+			}
+		});
+
+		saveButton.addActionListener(new ActionListener() {
+			public void actionPerformed(ActionEvent e) {
+				if (nameText.getText().isEmpty()) {
+					JOptionPane.showMessageDialog(null, "Please insert a rule name!");
+				} else {
+					// Add functionality here.
+				}
+			}
+		});
+
+		controlPanel.add(clearButton);
+		controlPanel.add(deleteButton);
+		controlPanel.add(saveButton);
+		return controlPanel;
+
+	}
+
+	/**
+	 * TODO: The currently * hard-coded values of the JBoxes are to be replaced with
+	 * the correct, final values once the ENUMs are created.
+	 * 
+	 * @return Returns the JPanel holding the line which allows users to add new
+	 *         metrics into the metrics list.
+	 */
+	public JPanel createAddMetricPanel() {
+		addNewMetricPanel.removeAll();
+		addNewMetricPanel.setLayout(new GridLayout(1, 6));
+		condition = new JComboBox<>();
+		setConditionVisibility();
+		JLabel ifCondition = new JLabel("IF", SwingConstants.CENTER);
+		JComboBox<String> value = new JComboBox<>();
+		value.addItem("LOC");
+		value.addItem("MET");
+
+		JComboBox<String> comparison = new JComboBox<>();
+		comparison.addItem(">");
+		comparison.addItem("<");
+		comparison.addItem("=");
+		comparison.addItem("!=");
+
+		JTextField threshold = new JTextField("10");
+
+		JButton addMetricButton = new JButton("Add");
+		addMetricButton.addActionListener(new ActionListener() {
+			public void actionPerformed(ActionEvent e) {
+				String metric;
+				try {
+					Integer.parseInt(threshold.getText());
+				} catch (NumberFormatException ex) {
+					JOptionPane.showMessageDialog(null, "Please check if your threshold input is correct!");
+				}
+				if (ruleMetrics.isEmpty()) {
+					metric = "IF " + value.getSelectedItem() + " " + comparison.getSelectedItem() + " "
+							+ threshold.getText();
+				} else {
+					metric = condition.getSelectedItem() + " " + value.getSelectedItem() + " "
+							+ comparison.getSelectedItem() + " " + threshold.getText();
+				}

```

Comment:
Instead of repeating ` + comparison.getSelectedItem() + " " + threshold.getText();` in both cases, you can have
```java
                                String baseMetric =  comparison.getSelectedItem() + " " + threshold.getText();
                                if (ruleMetrics.isEmpty()) {
  					metric = "IF " + value.getSelectedItem() + " " baseMetric;
  				} else {
  					metric = condition.getSelectedItem() + " " baseMetric;
  				}
```
or if you want to be supper short:
```java
String baseMetric =  comparison.getSelectedItem() + " " + threshold.getText();
String metric = ruleMetrics.isEmpty()
    ? "IF " + value.getSelectedItem() + " " baseMetric
    : condition.getSelectedItem() + " " baseMetric;
```